packages/ui/src/themes/buildTheme.ts

```diff
@@ -0,0 +1,33 @@
+import type { PartialTheme, ParsedElements } from '~/contexts/AppearanceContext';
+
+/**
+ * Given a complete theme object minus descriptors, returns a full ParsedElements object with generated descriptors.
+ */
+export function buildTheme(p: PartialTheme): ParsedElements {
+  const theme: Partial<ParsedElements> = {};
+
+  for (const descriptor in p) {
+    if (p[descriptor as keyof PartialTheme]) {
+      const { className, style } = p[descriptor as keyof PartialTheme]!;
+      theme[descriptor as keyof ParsedElements] = {
+        descriptor: `cl-${descriptor}`,
+        className: className ?? 'debug',
```

Comment:
This is default unnecessary after [this PR](https://github.com/clerk/javascript/pull/4022) right?

Reply:
Yup, that's in `main` so we'll pick it up once we're in #3976 which targets `main`.